src/cli/commands/deploy.ts

```diff
@@ -120,6 +138,33 @@ export async function deploy(options: SWACLIConfig) {
       logger.log(`  ${chalk.green(resolvedApiLocation)}`);
       logger.log(``);
     }
+    if (!apiLanguage) {
+      apiLanguage = DEFAULT_CONFIG.apiLanguage;
+      apiVersion = DEFAULT_CONFIG.apiVersion;
+      logger.log(`Consider providing api-language and version using --api-language and --api-version flags,
+      otherwise default values apiLanguage: ${apiLanguage} and apiVersion: ${apiVersion} will apply`);
+    } else if (apiLanguage && !apiVersion) {
+      switch (apiLanguage) {
+        case "node":
+          apiVersion = DEFAULT_NODE_VERSION;
+          break;
+        case "python":
+          apiVersion = DEFAULT_PYTHON_VERSION;
+          break;
+        case "dotnet":
+          apiVersion = DEFAULT_DOTNET_VERSION;
+          break;
+        case "dotnetisolated":
+          apiVersion = DEFAULT_DOTNET_ISOLATED_VERSION;
+          break;
+        default:
+          apiLanguage = DEFAULT_CONFIG.apiLanguage;
+          apiVersion = DEFAULT_CONFIG.apiVersion;
+          break;
+      }
+      logger.silly(`Api language "${apiLanguage}" is provided but api version is not provided.
+      Assuming default version "${apiVersion}"`);
+    }
```

Comment:
Some parts of this code logic look similar to `detect.ts#getDefaultVersion(). Can't we move this code into a helper function to avoid duplication?
`https://github.com/Azure/static-web-apps-cli/blob/772d0a050e185982d985a79e8d155d9d4525865a/src/core/frameworks/detect.ts#L398